src/usdb_syncer/song_txt/auxiliaries.py

```diff
@@ -42,14 +47,51 @@ def make_large_enough(self) -> int:
         return factor
 
 
-def replace_false_apostrophes_and_quotation_marks(value: str) -> str:
+def replace_false_apostrophes(value: str) -> str:
     # two single upright quotation marks ('') by double upright quotation marks (")
-    # grave accent (`), acute accent (´), prime symbol (′) and upright apostrophe (')
-    # by typographer’s apostrophe (’)
+    # grave (`) and acute accent (´), prime symbol (′), left single quotation mark (‘)
+    # and upright apostrophe (') by typographer’s apostrophe (’)
     return (
         value.replace("''", '"')
         .replace("`", "’")
         .replace("´", "’")
         .replace("′", "’")
+        .replace("‘", "’")
         .replace("'", "’")
     )
+
+
+def replace_false_quotation_marks(
+    value: str, language: str | None, marks_total: int
+) -> tuple[str, int, int]:
+    # replaces quotation marks with the language-specific counterparts
+    quotation_marks = next(
+        (
+            quote_marks
+            for langs, quote_marks in QUOTATION_MARKS.items()
+            if language in langs
+        ),
+        ("“", "”"),
+    )
+
+    quotation_mark_indices: list[int] = []
+    for mark in QUOTATION_MARKS_TO_REPLACE:
+        quotation_mark_indices.extend(
+            match.start() for match in re.finditer(re.escape(mark), value)
+        )
+    quotation_mark_indices.sort()
+
+    marks_fixed = 0
+    for str_index in quotation_mark_indices:
+        marks_index = marks_total % 2
```

Comment:
This doesn't take into account nested quotes. Not sure how common it would be in lyrics, but in theory you could have something like `he said 'hello "world"'`. So there is even a semantic difference between different kinds of quotation marks.

Reply:
Agreed, I guess in theory this could occur, even though I cannot recall having seen that even once. But indeed the current implementation would screw things up for that case.

Reply:
If we want to take that risk, maybe we should at least have a comment that this case is intentionally unsupported.